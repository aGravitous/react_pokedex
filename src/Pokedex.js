import React, { Component } from "react";
import Pokecard from "./Pokecard";
import pokecards from "./pokecards";

class Pokedex extends Component {
 render () {

    return (
        <div className="Pokedex">
            { pokecards.map(card =>  <Pokecard key={card.id} name={ card.name }
            id={ card.id } type={ card.type }/>) }
        </div>
    );
 }
};

export default Pokedex;