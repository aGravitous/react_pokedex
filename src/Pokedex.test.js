import React from "react";
import { shallow, mount } from "enzyme";
import Pokedex from "./Pokedex";
import toJson from "enzyme-to-json";


// smoke test
it("renders without crashing", function() {
  shallow(<Pokedex />);
});

// mount smoke test
it("renders using mount", function() {
    mount(<Pokedex />);
  });

// snapshot test
it("matches snapshot", function() {
    let wrapper = shallow(<Pokedex />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });  