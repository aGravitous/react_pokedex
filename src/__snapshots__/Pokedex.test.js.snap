// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Pokedex"
>
  <Pokecard
    id={4}
    key="4"
    name="Charmander"
    type="fire"
  />
  <Pokecard
    id={7}
    key="7"
    name="Squirtle"
    type="water"
  />
  <Pokecard
    id={11}
    key="11"
    name="Metapod"
    type="bug"
  />
  <Pokecard
    id={12}
    key="12"
    name="Butterfree"
    type="flying"
  />
  <Pokecard
    id={25}
    key="25"
    name="Pikachu"
    type="electric"
  />
  <Pokecard
    id={39}
    key="39"
    name="Jigglypuff"
    type="normal"
  />
  <Pokecard
    id={94}
    key="94"
    name="Gengar"
    type="poison"
  />
  <Pokecard
    id={133}
    key="133"
    name="Eevee"
    type="normal"
  />
</div>
`;
